src/bam/mod.rs

```diff
@@ -92,14 +92,71 @@ pub trait Read: Sized {
     /// # Returns
     ///
     /// `Ok(true)` if record was read without error, Ok(false) if there is no more record in the file.
+    ///
+    /// Example:
+    /// ```
+    /// use rust_htslib::bam::{Error, Read, IndexedReader, Record};
+    ///
+    /// let mut bam = IndexedReader::from_path(&"test/test.bam").unwrap();
+    /// bam.fetch((0, 1000, 2000)); // reads on tid 0, from 1000bp to 2000bp
+    /// let mut record = Record::new();
+    /// loop {
+    ///     match bam.read(&mut record){
+    ///         Ok(true) => {
+    ///             println!("Read sequence: {:?}", record.seq().as_bytes());
+    ///         },
+    ///         Ok(_) => break, //no more reads
+    ///         Err(_) => panic!("BAM parsing failed...")
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// Consider using [`rc_records`](#tymethod.rc_records) instead.
     fn read(&mut self, record: &mut record::Record) -> Result<bool>;
 
     /// Iterator over the records of the seeked region.
     /// Note that, while being convenient, this is less efficient than pre-allocating a
     /// `Record` and reading into it with the `read` method, since every iteration involves
     /// the allocation of a new `Record`.
+    ///
+    /// This is about 10% slower than record in micro benchmarks.
+    ///
+    /// Consider using [`rc_records`](#tymethod.rc_records) instead.
     fn records(&mut self) -> Records<'_, Self>;
 
+    /// Records iterator using an Rc to avoid allocating a Record each turn.
+    /// This is about 1% slower than the [`read`](#tymethod.read) based API in micro benchmarks,
+    /// but has nicer ergonomics (and might not actually be slower in your applications).
+    ///
+    /// Example:
+    /// ```
+    /// use rust_htslib::bam::{Error, Read, Reader, Record};
+    /// use rust_htslib::htslib; // for BAM_F*
+    /// let mut bam = Reader::from_path(&"test/test.bam").unwrap();
+    /// let record = Record::new();
+    //
```

Comment:
What I meant was that this `record` isn't used anywhere, so can be omitted completely :D
Aaaaand we lost a slash there

Reply:
D'oh.